### DISCUSSION TITLE
sizes in units

### DISCUSSION THREAD

**Discussion by kedXD** [primefaces/community#3232] [Figma UI Kit]
Is there any reason behind going for sizes in decimals? 
Ex. fonts are like sm 12.25, lg 15.75 and padding 8.75 and 5.25 etc. Shouldn't it be expressed in numbers like 12, 6, 8 etc?

**Comment by onursenture** [comment]
Hello, this is because we want the font sizes we use in libraries to match the font sizes in the UI Kit. Since we use REM, this situation applies to font sizes, spacing, and sizing values. In summary: Because the base font size we use in UI Libraries is 14px, it's likely that the resulting sizes will have decimal values.

**Comment by P3ek** [comment]
Forgive my ignorance, but what is the most effective way to update our kit to work in multiples of 2/4/8 etc? 
I've read that [updating the root-level font size to 16px](https://github.com/orgs/primefaces/discussions/971) is a way to do this, but the Figma UI kit 3.0.1 already has Base font size as 16px in token studio settings?

**Reply by onursenture** [reply]
In Tokens Studio Settings, you must link the base font size to the app.font.size token. It is actually 14px. So, currently use 14px as the base font size in the updated Figma UI Kit. You need to set it in the Tokens Studio settings.

<img width="417" alt="image" src="https://github.com/user-attachments/assets/97bb213d-edc9-4fd8-b9bf-43942296d344" />

After this, you can switch to any desired base font size by changing the app.font.size token.

**Reply by P3ek** [reply]
Perfect, changed {app.font.size} to 16 after updating token studio settings. Exactly what I needed. Big thanks @onursenture .

**Comment by designer-cebanas** [comment]
Hello @onursenture, thank you for your reply. I have changed the app.font.size to 16, once I apply to the file, all the variables disconnect in Figma, is there a way that we can switch to app.font.size 16, but keep the variables connected to all elements?

**Reply by onursenture** [reply]
Hi there! After making your changes, did you use Tokens Studio’s “Export to Figma” feature with the exact settings shown in the two example screens I shared? If that doesn’t work, you might try rerunning the export without changing anything else. According to the Tokens Studio documentation, re-exporting can sometimes resolve connection issues. https://docs.tokens.studio/variables-and-styles/export-to-figma-options-menu

<img width="390" alt="image" src="https://github.com/user-attachments/assets/b7af2901-cde1-4d95-9b5a-26fa12bbf1f9" />
<img width="387" alt="image" src="https://github.com/user-attachments/assets/cf135e62-20bf-4140-b3bc-a39329ae8d3f" />


Additionally, keep in mind that Figma doesn’t currently support multiple-value tokens. For example, if a spacing token in Tokens Studio has a value like “0.5rem 0.75rem,” it will be skipped during export and won’t appear as a Figma Variable. Make sure to double-check for these cases when reviewing your tokens. You can find more details in their docs: https://docs.tokens.studio/variables-and-styles/export-to-figma-skipped-variables
